### PR TITLE
Hotfix for explosion knockback issues

### PR DESCRIPTION
**Pull request by mathiascode**
Hotfix for #4575

### CODE REVIEW

File: src/ChunkMap.cpp [hunk 1]
```diff
@@ -1313,21 +1313,14 @@ void cChunkMap::DoExplosionAt(double a_ExplosionSize, double a_BlockX, double a_
...
-
-				// Exposure reduced by armor
-				EntityExposure = EntityExposure * (1.0f - a_Entity.GetEnchantmentBlastKnockbackReduction());
+			// Hotfix for #4575
+			// TODO: Enhance knockback to be more vanilla-like. Reapply blast reduction.
+			auto Knockback = (1 / std::max(1.0, DistanceFromExplosion.Length())) * 6 * ExplosionSizeInt;
 
-				double Impact = (1 - ((Length / ExplosionSizeInt) / 2)) * EntityExposure;
+			DistanceFromExplosion.Normalize();
+			DistanceFromExplosion = Vector3d(DistanceFromExplosion.x * Knockback, DistanceFromExplosion.y + (Knockback * 0.3), DistanceFromExplosion.z * Knockback);
```
Comment (bearbin) [inline src/ChunkMap.cpp]: Vector3s support vector multiplication, so no need to decompose and recompose.

```suggestion
			DistanceFromExplosion *= Vector3d(1, 0.3, 1) * Knockback;
```

Comment (mathiascode) [reply src/ChunkMap.cpp]: This breaks height knockback (notice the +).

File: src/ChunkMap.cpp [hunk 2]
```diff
@@ -1313,21 +1313,14 @@ void cChunkMap::DoExplosionAt(double a_ExplosionSize, double a_BlockX, double a_
 				a_Entity.TakeDamage(dtExplosion, nullptr, static_cast<int>((1 / std::max(1.0, DistanceFromExplosion.Length())) * 8 * ExplosionSizeInt), 0);
 			}
 
-			double Length = DistanceFromExplosion.Length();
-			if (Length <= ExplosionSizeInt)  // Entity is impacted by explosion
-			{
-				float EntityExposure = a_Entity.GetExplosionExposureRate(ExplosionPos, static_cast<float>(a_ExplosionSize));
-
-				// Exposure reduced by armor
-				EntityExposure = EntityExposure * (1.0f - a_Entity.GetEnchantmentBlastKnockbackReduction());
```
Comment (peterbell10) [inline src/ChunkMap.cpp]: Why remove `EntityExposure`?

Comment (mathiascode) [reply src/ChunkMap.cpp]: It seems to be broken (https://github.com/cuberite/cuberite/issues/4527), and I'm not sure how to fix it/how to incorporate it.

Comment (peterbell10) [reply src/ChunkMap.cpp]: `GetExplosionExposureRate` scales by the fraction of the entities bounding box that overlaps with the explosion, so should only be < 1 if the entity is on the boundary of the explosion.

You've also broken blast protection's effect on knockback.

Comment (mathiascode) [reply src/ChunkMap.cpp]: If you know this code better, then feel free to make contributions or take over this PR. I didn't really want to work on it, but I don't want to leave users exposed to this bug for too long either.

**Pull request closed without merging**